embedded-graphics/src/mock_display.rs

```diff
@@ -1,24 +1,125 @@
-use crate::drawable::{Dimensions, Pixel};
+use crate::drawable::Pixel;
 use crate::pixelcolor::BinaryColor;
 use crate::prelude::*;
-use crate::{Drawing, SizedDrawing};
+use crate::Drawing;
+use core::{
+    cmp::PartialEq,
+    fmt::{self, Write},
+    iter,
+};
+
+const SIZE: usize = 64;
 
 /// Mock display for use in tests and some doc examples. Do not use directly!
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
-pub struct MockDisplay<C>(pub [[C; 24]; 16]);
+#[derive(Clone)]
+pub struct MockDisplay<C>([Option<C>; SIZE * SIZE])
+where
+    C: PixelColor;
+
+impl<C> MockDisplay<C>
+where
+    C: PixelColor,
+{
+    /// Creates a new empty mock display.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Returns a iterator over all pixels on the display.
+    pub fn pixels<'a>(&'a self) -> impl Iterator<Item = Option<C>> + 'a {
```

Comment:
Nit: This method can return `impl Iterator<Item = &'a Option<C>>`. The copy here doesn't really matter for tests of course, but it feels cleaner to me to return a reference.

This change introduces an error in `font6x8.rs` which can be fixed by adding `.cloned()` there. Like this:

```rust
assert!(inverted_display_normal.eq(display_inverse.pixels().copied()));
```

Reply:
To be more useful `pixels` should have returned `Pixel<...>` anyway, but for now I removed it completely and replaced it by `get_pixel` and `set_pixel`. If we need this iterator later it's easy to add it again.